app/_how-tos/configure-oidc-with-consumers.md

```diff
@@ -0,0 +1,166 @@
+
+faqs:
+  - q: When using Consumer authorization, is Consumer mapping required?
+    a: |
+      Consumer mapping is required by default, but you can make Consumer mapping optional and non-authorizing by setting the OpenID Connect plugin's configuration parameter [`config.consumer_optional`](/plugins/openid-connect/#schema--config-consumer-optional) to `true`.
+  - q: Can I use Consumer mapping with ACL allow/deny lists?
+    a: |
+      Yes, you can combine the Consumer claim with the authorization group claim to further secure your environment. See the how-to on [configuring OIDC with ACL groups](/how-to/configure-oidc-with-acl-auth/) for more information.
+
+cleanup:
+  inline:
+    - title: Clean up Konnect environment
+      include_content: cleanup/platform/konnect
+      icon_url: /assets/icons/gateway.svg
+    - title: Destroy the {{site.base_gateway}} container
+      include_content: cleanup/products/gateway
+      icon_url: /assets/icons/gateway.svg
+
+---
+
+## 1. Enable the OpenID Connect plugin
+
+Using the Keycloak and {{site.base_gateway}} configuration from the [prerequisites](#prerequisites), 
+set up an instance of the OpenID Connect plugin. In this example, we're using the simple password grant with the `preferred_username` Consumer claim.
+
+Enable the OpenID Connect plugin on the `example-service` Service:
+
+{% entity_examples %}
+entities:
+  plugins:
+    - name: openid-connect
+      service: example-service
+      config:
+        issuer: ${issuer}
+        client_id:
+        - ${client-id}
+        client_secret:
+        - ${client-secret}
+        client_auth:
+        - client_secret_post
+        auth_methods:
+        - password
+        consumer_claim:
+        - preferred_username
+        consumer_by:
+        - username
+variables:
+  issuer:
+    value: $ISSUER
+  client-id:
+    value: $CLIENT_ID
+  client-secret:
+    value: $CLIENT_SECRET
+{% endentity_examples %}
+
+In this example:
+* `issuer`, `client ID`, `client secret`, and `client auth`: Settings that connect the plugin to your IdP (in this case, the sample Keycloak app).
+* `auth_methods`: Password grant, for easy testing.
+* `consumer_claim` and `consumer_by` : Looks for a `preferred_username` in the token payload and maps it to the Consumer entity by the entity's `username` value.
+
+{% include_cached plugins/oidc/client-auth.md %}
+
+## 2. Create a Consumer
+
+First, let's try to access the Service without a matching Consumer.
+Request the Service with the basic authentication credentials created in the [prerequisites](#prerequisites):
+
+{% validation request-check %}
+url: /anything
+method: GET
+status_code: 403
+user: "john:doe"
+display_headers: true
+{% endvalidation %}
+
+You should get a `403 Forbidden` error code, which means the Service is protected by authentication.
+
+Create a Consumer with a username that matches the user in your IdP, in this case `john`:
```

Comment:
Can you update the keycloak prereq file (keycloak-password) so that we set the specific username there? Otherwise, I'd have to copy and edit the decK file config here. Also, we should use one of the names from this list: https://developers.google.com/style/examples?hl=en#example-person-names